bundles/org.openhab.binding.enigma2/src/main/java/org/openhab/binding/enigma2/internal/Enigma2Client.java

```diff
@@ -0,0 +1,350 @@
+ * 
+ * @see <a href=
+ *      "https://github.com/E2OpenPlugins/e2openplugin-OpenWebif/wiki/OpenWebif-API-documentation">OpenWebif-API-documentation</a>
+ *
+ * @author Guido Dolfen - Initial contribution
+ */
+@NonNullByDefault
+public class Enigma2Client {
+    private final Logger logger = LoggerFactory.getLogger(Enigma2Client.class);
+
+    static final String PATH_REMOTE_CONTROL = "/web/remotecontrol?command=";
+    static final String PATH_POWER = "/web/powerstate";
+    static final String PATH_VOLUME = "/web/vol";
+    static final String PATH_SET_VOLUME = "/web/vol?set=set";
+    static final String PATH_TOGGLE_MUTE = "/web/vol?set=mute";
+    static final String PATH_TOGGLE_POWER = "/web/powerstate?newstate=0";
+    static final String PATH_MESSAGE = "/web/message?type=";
+    static final String PATH_ALL_SERVICES = "/web/getallservices";
+    static final String PATH_ZAP = "/web/zap?sRef=";
+    static final String PATH_CHANNEL = "/web/subservices";
+    static final String PATH_EPG = "/web/epgservicenow?sRef=";
+    static final String PATH_ANSWER = "/web/messageanswer?getanswer=now";
+    static final int TYPE_QUESTION = 0;
+    static final int TYPE_INFO = 1;
+    static final int TYPE_WARNING = 2;
+    static final int TYPE_ERROR = 3;
+    private final Map<String, @Nullable String> channels = new ConcurrentHashMap<>();
+    private final String host;
+    private boolean power;
+    private String channel = "";
+    private String title = "";
+    private String description = "";
+    private String answer = "";
+    private int volume = 0;
+    private boolean mute;
+    private boolean online;
+    private boolean initialized = false;
+    private boolean asking = false;
+    private LocalDateTime lastAnswerTime = LocalDateTime.of(2020, 1, 1, 0, 0); // Date in the past
+    private Enigma2HttpClient enigma2HttpClient;
+
+    public Enigma2Client(String host, @Nullable String user, @Nullable String password, int requestTimeout) {
+        this.enigma2HttpClient = new Enigma2HttpClient(requestTimeout);
+        if (StringUtils.isNotEmpty(user) && StringUtils.isNotEmpty(password)) {
+            this.host = "http://" + user + ":" + password + "@" + host;
+        } else {
+            this.host = "http://" + host;
+        }
+    }
+
+    public boolean refresh() {
+        boolean wasOnline = online;
+        refreshPower();
+        if (!wasOnline && online) {
+            // Only refresh all services if the box changed from offline to online and power is on
+            // because it is a performance intensive action.
+            refreshAllServices();
+        }
+        refreshChannel();
+        refreshEpg();
+        refreshVolume();
+        refreshAnswer();
+        return online;
+    }
+
+    public void refreshPower() {
+        Optional<Document> document = transmitWithResult(PATH_POWER);
+        if (document.isPresent()) {
+            online = true;
```

Comment:
Is there a reason to have `power `separate from `online`. They seem to be highly correlated?

Reply:
There is a different between power an online. Power is off if the thing is in standby, but the it is still online, because tehre is an available HTTP-Connection an the binding is able to change the power state. If the thing is offline, there is no available HTTP-Connection.